compiler/rustc_metadata/src/creader.rs

```diff
@@ -979,13 +980,53 @@ impl<'a, 'tcx> CrateLoader<'a, 'tcx> {
         }
     }
 
+    fn report_future_incompatible_deps(&self, krate: &ast::Crate) {
+        let name = self.tcx.crate_name(LOCAL_CRATE);
+
+        if name.as_str() == "wasm_bindgen" {
+            if !env::var("CARGO_PKG_VERSION_MAJOR")
+                .ok()
+                .and_then(|major| u64::from_str(&major).ok())
+                .is_some_and(|major| major == 0)
+            {
+                return;
+            }
+            if !env::var("CARGO_PKG_VERSION_MINOR")
+                .ok()
+                .and_then(|minor| u64::from_str(&minor).ok())
+                .is_some_and(|minor| minor <= 2)
+            {
+                return;
+            }
+            if !env::var("CARGO_PKG_VERSION_PATCH")
+                .ok()
+                .and_then(|patch| u64::from_str(&patch).ok())
+                .is_some_and(|minor| minor <= 87)
+            {
+                return;
+            }
```

Comment:
Unfortunately, AFAIK, rustc doesn't have any access to dependency versions, so accessing it through `CARGO_PKG_VERSION_*` was used as a workaround.

Thanks to [bjorn3](https://github.com/bjorn3/) for the suggestion.

Reply:
This should mention in a code comment that version 0.1.88 does not exist and never will.

Reply:
Did you mean v0.**2**.88 because it is yanked?
Yanked crates exist and can be used, I think they just require opt-in by users who don't have them in their `Cargo.lock`.

If you actually meant v0.1.88 I'm unsure how this is relevant here.

Reply:
I meant 0.1.88, because it would make the version comparison here fail and claim the crate didn't need an update.

Reply:
I see now, this is a bug really, I will fix it.

Reply:
Thank you, addressed.